docs_dev/kubeflow_deployment.md

```diff
@@ -0,0 +1,282 @@
+
+  	 ```
+  	 curl -L -O /tmp/source.tar.gz https://github.com/kubeflow/kubeflow/archive/v0.2.0.tar.gz
+     tar -zxvf /tmp/source.tar.gz
+	 ```
+
+  1. Add the registry as a file path and not a git registry
+
+  	 ```
+  	 ks registry add kubeflow $/tmp/source
+  	 ```
+
+  * Since the registry is added from a local file and not Git the registry will only
+    be valid on that machine. To mitigate the effects of moving the ksonnet app to other machines you can install any packages you think users might want
+
+  * Installed packages are stored in the ksonnet app vendor directory; they will be
+    valid on other machines provided you preserve the vendor directory (e.g. check it in to source control).
+
+  * There is an open issue [ksonnet/ksonnet#64](https://github.com/ksonnet/ksonnet/issues/641) to support this pattern
+    natively in ksonnet and address these limitations.
+
+## Open Question: Ensuring a common experience
+
+An open question is finding the right balance between ensuring a common experience across all platforms
+while still giving maintainers of a particular platform the ability to customize that experience.
+
+I think we'd like to have a well defined concept of base Kubeflow which consists of a good set of components
+that we think all users will want. Platform maintainers should be discouraged from removing components in base.
+If users want to remove certain base components they are free to do so and can easily do so.
+
+We have yet to find a good method for defining the base components. Currently, the base components
+is defined by the components generated in the deploy script. This works but will probably be difficult
+to avoid fragmentation as scripts are added for more platforms.
+
+We had previously tried defining a ksonnet core prototype consisting of all the components that should be 
+installed by default. This had some unfortunate side effects and I think we want to avoid creating a single prototype
+for multiple components going forward
+
+   * A user should be able to manage the components individually; e.g. updating JupyterHub without updating other components
+   * We end up combining the parameters for all the components into a single prototype file which is a bit unwieldy
+   * It limits our ability to organize components into ksonnet packages by application (e.g. TensorFlow, PyTorch, Jupyter)
+
+## Monitoring Kubeflow Deployment
+
+Monitoring deployment is an open area. The initial focus is:
+
+  1. Informing the user when Kubeflow is fully deployed and ready to use
+  1. Surfacing relevant events (e.g. as K8s events)
+
+For more info please take a look at some of the open issues and consider
+opening more. 
+
+The current thinking is to follow the guidance of sig-apps and use an [application resource](https://github.com/kubernetes-sigs/application)
+to represent Kubeflow and attach events, status, and other metrics to that application as appropriate.
```

Comment:
In addition to the application crd noted above - could the [SecurityProfile](https://github.com/easeway/community/blob/a8cb2060dc621664c86b185c7426367994b181b5/keps/draft-20180418-security-profile.md) also be leveraged?

Reply:
Assuming you mean for security purposes (as opposed to monitoring) then I think so.  It looks like SecurityProfile is still very early though and not even supported by sig-apps yet.

Reply:
yes. It is early but it seems like it could be combined with application resource - where a UI could allow different components and security profiles to be bundled and deployed. 

Is the bootstrapper assuming that only one kubeflow instance would be deployed or many? EG:
- kubeflow-admin
above privileged namespace could deploy kubeflow 'instances' like below and have different directories (that could include gitops) for each deployment
- kubeflow-teamX
- kubeflow-datascientistY
- kubeflow-pytorch-gpu-1